cclib/method/moments.py

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2018, the cclib development team
+#
+# This file is part of cclib (http://cclib.github.io) and is distributed under
+# the terms of the BSD 3-Clause License.
+
+"""Calculation of electric multipole moments based on data parsed by cclib."""
+
+import numpy
+
+from cclib.parser.utils import convertor
+from cclib.method.calculationmethod import Method
+
+
+class Moments(Method):
+    def __init__(self, data):
+        self.required_attrs = ('atomcoords', 'atomcharges')
+
+        super(Moments, self).__init__(data)
+
+    def __str__(self):
+        """Returns a string representation of the object."""
+        return "Multipole moments of %s" % (self.data)
+
+    def __repr__(self):
+        """Returns a representation of the object."""
+        return 'Moments("%s")' % (self.data)
+
+    def _calculate_dipole(self, charges, coords, origin):
+        coords_au = convertor(coords, 'Angstrom', 'bohr')
+        
+        if self.data.charge == 0:
+            dipole = numpy.dot(charges, coords_au)
+        else:
+            origin_au = convertor(origin, 'Angstrom', 'bohr')
+            dipole = numpy.dot(charges, coords_au - origin_au)
+            
+        return convertor(dipole, 'ebohr', 'Debye')
+
+    def _calculate_quadrupole(self, charges, coords, origin):
```

Comment:
Even though this is private, please add a docstring. It should include a comment that this is the _traceless_ quadrupole tensor.